src/i2c/src/i2c.rs

```diff
@@ -0,0 +1,548 @@
+// I2C backend device
+//
+// Copyright 2021 Linaro Ltd. All Rights Reserved.
+//          Viresh Kumar <viresh.kumar@linaro.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use libc::{c_ulong, ioctl, EADDRINUSE, EADDRNOTAVAIL, EINVAL};
+use std::fs::{File, OpenOptions};
+use std::os::unix::io::AsRawFd;
+use vmm_sys_util::errno::{errno_result, Error, Result};
+
+// The type of the `req` parameter is different for the `musl` library. This will enable
+// successful build for other non-musl libraries.
+#[cfg(target_env = "musl")]
+type IoctlRequest = libc::c_int;
+#[cfg(not(target_env = "musl"))]
+type IoctlRequest = c_ulong;
+
+/// Linux I2C/SMBUS definitions
```

Comment:
Ioctl codes for Linux I2C/SMBUS from file include/uapi/linux/i2c-dev.h?

Reply:
> Ioctl codes for Linux I2C/SMBUS from file include/uapi/linux/i2c-dev.h?

Right. Its all from there upto line 91 here.